### PR TITLE
CLDR-13630 Fix verification chart for numbers

### PR DESCRIPTION
**Pull request by macchiati**
##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13630
- [ ] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

**Comment by macchiati** [comment]
Refactored code and created test in the process of debugging. 

Actual fixes were:

- Skip passing "0" values to ICU.
- Fixed case where raw ICU number formatter was being used instead of creating

**Comment by macchiati** [comment]
the chart https://www.unicode.org/repos/cldr-aux/charts/37/verify/numbers/it.html shows fixed values.

File: tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCompactNumbers.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        CompactDecimalFormat cdfs = BuildIcuCompactDecimalFormat.build(cldrFile, debugCreationErrors,
+            debugOriginals, CompactStyle.LONG, locale2, CurrencyStyle.PLAIN, currencyCode);
+        CompactDecimalFormat cdfCurr = BuildIcuCompactDecimalFormat.build(cldrFile, debugCreationErrors,
+            debugOriginals, CompactStyle.SHORT, locale2, CurrencyStyle.CURRENCY, currencyCode);
+
+        Set<Double> allSamples = VerifyCompactNumbers.collectSamplesAndSetFormats(currencyCode, locale, SDI, cdf, cdfs, cdfCurr);
+
+        for (double source : allSamples) {
+            if (false && source == 22000000 && locale.equals("cs")) {
+                System.out.println("**");
```
Comment (pedberg-icu) [inline tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCompactNumbers.java]: What is the problem with "cs" compact decimals? Is there a ticket?

Comment (macchiati) [reply tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCompactNumbers.java]: That's just debugging code. (there a bunch of that that could be cleaned up in various areas, but I'd rather not try to address in v37)

**Pull request merged**